backend/typescript/services/implementations/volunteerService.ts

```diff
@@ -148,6 +164,25 @@ class VolunteerService implements IVolunteerService {
     return userVolunteerDTOs;
   }
 
+  async getCheckInsAndSchedules(
+    volunteerId: string,
+  ): Promise<(CheckInDTO | SchedulingDTO)[]> {
+    const checkIns = await (
+      await this.checkInService.getCheckInsByVolunteerId(volunteerId)
+    ).map((checkIn) => ({ ...checkIn, type: ShiftType.CHECKIN }));
+    const schedulings = await (
+      await this.schedulingService.getSchedulingsByVolunteerId(volunteerId)
+    ).map((scheduling) => ({ ...scheduling, type: ShiftType.SCHEDULING }));
+    const shifts = [...checkIns, ...schedulings].sort((a, b) => {
+      const date1: Date =
+        "startDate" in a ? new Date(a.startDate) : new Date(a.startTime);
+      const date2: Date =
+        "startDate" in b ? new Date(b.startDate) : new Date(b.startTime);
+      return date2.valueOf() - date1.valueOf();
```

Comment:
will have to convert to dayjs for comparison but 
```suggestion
      return date2.isBefore(date1)
```